ocaml/xenopsd/xc/device.ml

```diff
@@ -816,14 +816,14 @@ module Vif = struct
               50000L
             (* 50ms by default *)
           in
-          let bytes_per_interval =
-            ((kbytes_per_s ^* 1024L) ^* timeslice_us) ^/ 1000000L
+          let bits_per_interval =
+            ((kbps ^* 1000L) ^* timeslice_us) ^/ 1000000L
```

Comment:
Please explain in the commit message the change of unit base (from 2 to 10) as well

Reply:
mh I did assume that this was being passed to `ip link` tbh. @edwintorok can we easily confirm that's the case? 👀 

It does look like we're just writing it to `xenstore` here so not sure where it's picked up :/

Reply:
This is not related to the `set_vf_rate` function in networkd... That one is for SR-IOV  virtual functions (VFs). This code here is setting values in xenstore for netback.

Reply:
See L950 and below in this file for the SR-IOV/VF case.

Reply:
Also, we could check what netback really does, but this code explicitly talks about _kbytes_ per second, not kbits, and I have no reason to believe this is wrong. Other than the fact that the XenAPI parameter is unfortunately called `kbps`... :/ The XS docs at https://docs.xenserver.com/en-us/xenserver/8/networking/manage also talk about bytes.

Reply:
Using `kbps` to mean kilobytes/s is very confusing, `kbps` isn't ideal but almost everywhere it means `Kbit/s` (with K power of 10).

I think we should deprecate that parameter (accept it as kilobytes for backward compat) and add a properly named parameter. Maybe `bitrate`? And then the CLI and XC can do whatever unit conversions they want but the API will have the full value (although obviously we cannot actually consistently set very small values, so the granularity might be confusing).